compiler/optimizer/OMRSimplifierHandlers.cpp

```diff
@@ -9962,29 +9968,55 @@ TR::Node *lconstSimplifier(TR::Node * node, TR::Block * block, TR::Simplifier *
    return node;
    }
 
-TR::Node *iabsSimplifier(TR::Node * node, TR::Block * block, TR::Simplifier * s)
+TR::Node *ilfdabsSimplifier(TR::Node * node, TR::Block * block, TR::Simplifier * s)
    {
    simplifyChildren(node, block, s);
-   auto child = node->getChild(0);
-   if (child->isNonNegative() &&
-       performTransformation(s->comp(), "%sSimplify iabs of non-negative child at [" POINTER_PRINTF_FORMAT "]\n",
-             s->optDetailString(), node))
+
+   static bool AllowFlagBasedTransformationForFloatOrDouble = feGetEnv("TR_AllowFlagBasedTransformationForFloatOrDouble") != NULL;
+
+   TR::ILOpCodes abs = node->getOpCodeValue();
+   TR::ILOpCodes neg = TR::BadILOp;
+   bool performFlagBasedTransformation = true;
```

Comment:
I think this bool is misnamed - it would be better as isFlagBasedTransformation defaulting to false and set to true. You would then later test AllowFlagBasedTransformationForFloatOrDobule and this condition. This is functionally correct, just harder to read.

Reply:
Good point, updated.